lib/src/Uncertainty/Distribution/openturns/ConditionedDistribution.hxx

```diff
@@ -0,0 +1,212 @@
+  Bool isContinuous() const override;
+
+  /** Check if the distribution is discrete */
+  Bool isDiscrete() const override;
+
+  /** Check if the distribution is integral */
+  Bool isIntegral() const override;
+
+  /** Get the support of a discrete distribution that intersect a given interval */
+  using DistributionImplementation::getSupport;
+  Sample getSupport(const Interval & interval) const override;
+
+  /** Get the standard deviation of the distribution */
+  Point getStandardDeviation() const override;
+
+  /** Get the skewness of the distribution */
+  Point getSkewness() const override;
+
+  /** Get the kurtosis of the distribution */
+  Point getKurtosis() const override;
+
+  /** Parameters value accessors */
+  void setParameter(const Point & parameter) override;
+  Point getParameter() const override;
+
+  /** Parameters description accessor */
+  Description getParameterDescription() const override;
+
+  /** Get the PDF gradient of the distribution */
+  using DistributionImplementation::computePDFGradient;
+  Point computePDFGradient(const Point & point) const override;
+
+  /** Get the CDF gradient of the distribution */
+  using DistributionImplementation::computeCDFGradient;
+  Point computeCDFGradient(const Point & point) const override;
+
+  /** Method save() stores the object through the StorageManager */
+  void save(Advocate & adv) const override;
+
+  /** Method load() reloads the object from the StorageManager */
+  void load(Advocate & adv) override;
+
+  /* Interface specific to ConditionedDistribution */
+
+  /** Conditioned distribution accessor */
+  Distribution getDistribution() const;
+
+  /** Probability table accessor */
+  Indices getConditioningIndices() const;
+  Point getConditioningValues() const;
+
+  /** Get the simplified version (or clone the distribution) */
+  Distribution getSimplifiedVersion() const;
+
+protected:
+
+private:
+  /** Expand the given marginal point to the underlying distribution argument point */
+  Point expandPoint(const Point & point) const;
+
+  /* Get the simplified version */
+  Bool hasSimplifiedVersion(Distribution & simplified) const;
+
+  /** Compute the mean of the distribution */
+  void computeMean() const override;
+
+  /** Compute the covariance of the distribution */
+  void computeCovariance() const override;
+
+  /** Compute the numerical range of the distribution given the parameters values */
+  void computeRange() override;
+
+  /** Update byproducts */
+  void update();
+
+  // main attributes
+  Distribution distribution_;
+  Indices conditioningIndices_;
+  Point conditioningValues_;
+
+  // cache some byproducts
+  Indices nonConditioningIndices_;
+  Distribution marginalConditionedDistribution_;
+  Scalar normalizationFactor_ = 1.0;
+
+  // simplified instance
+  Distribution simplifiedVersion_;
+  Bool useSimplifiedVersion_ = false;
+
+  // for CDF integration
+  IntegrationAlgorithm integrationAlgorithm_;
```

Comment:
If the integration algorithm is too costly (e.g. if the dimension is large), this might prevent us from using this class. I would suggest to provide a way to use a sampling based quadrature method, e.g. Monte-Carlo. This was done at:

https://github.com/openturns/otbenchmark/blob/ae19ef309f7d75ee249687f0802331f8b84f3c87/otbenchmark/_ConditionalDistribution.py#L216

The Python implementation was, however, not very tight, because the integration algorithms were located in two separate branches of the code: factoring the code in a single location is much better. This could be possible if we could create a `IntegrationAlgorithm` based on a `ExperimentIntegration`. But this is not possible, because `IntegrationAlgorithm` is in src/Base, while `ExperimentIntegration` is in src/Uncertainty. In the meantime, the current code is, indeed, the best option. +1

Reply:
I tried this but unfortunately it returns incorrect results even in low dimension

Reply:
@jschueller Do you have the problematic examples somewhere?

Reply:
absolutely, I've just retried on the NormalCopula example from the test:
[pcmc.patch.txt](https://github.com/user-attachments/files/17282563/pcmc.patch.txt)

Reply:
beware I forgot the normalization term, also we should probably use QMC

Reply:
How can you say that computeCDF is wrong? I tried your branch with the patch, and the CDF looks good to me, at least in 2D.
```python
import openturns as ot
import openturns.experimental as otexp
import openturns.viewer as otv

dim = 3
R = ot.CorrelationMatrix(dim)
R[1, 0] = R[2, 0] = R[2, 1] = 0.9
copula = ot.NormalCopula(R)
distribution = otexp.PointConditionalDistribution(copula, [1], [0.9])
print("range=", distribution.getRange())
print("CDF along the diagonal")
for i in range(11):
    x = i / 10.0
    print("CDF(", x, ",", x, ")=", distribution.computeCDF([x, x]))
print("CDF along the anti-diagonal")
for i in range(11):
    x = i / 10.0
    print("CDF(", x, ",", 1-x, ")=", distribution.computeCDF([x, 1-x]))
otv.View(distribution.drawCDF([0.0]*2, [1.0]*2)).save("cdf.png")
```
produces
```
[regis@localhost Conditionnel]$ PYTHONPATH=$PWD/openturns/build/install/lib64/python3.10/site-packages/ python test_conditioned_cdf.py 
range= [0, 1]
[0, 1]
CDF along the diagonal
CDF( 0.0 , 0.0 )= 0.0
CDF( 0.1 , 0.1 )= 7.123771367763459e-12
CDF( 0.2 , 0.2 )= 1.0668398787525147e-08
CDF( 0.3 , 0.3 )= 9.171281985660072e-07
CDF( 0.4 , 0.4 )= 2.4132986226664257e-05
CDF( 0.5 , 0.5 )= 0.00033062506177891654
CDF( 0.6 , 0.6 )= 0.0029900631848383597
CDF( 0.7 , 0.7 )= 0.020188394310581826
CDF( 0.8 , 0.8 )= 0.10808165475328041
CDF( 0.9 , 0.9 )= 0.45218826893692965
CDF( 1.0 , 1.0 )= 1.0
CDF along the anti-diagonal
CDF( 0.0 , 1.0 )= 0.0
CDF( 0.1 , 0.9 )= 1.1604021917018248e-08
CDF( 0.2 , 0.8 )= 2.2650283596192996e-06
CDF( 0.3 , 0.7 )= 4.206875021598631e-05
CDF( 0.4 , 0.6 )= 0.00020143028491280865
CDF( 0.5 , 0.5 )= 0.00033062506177891654
CDF( 0.6 , 0.4 )= 0.00020143028491280903
CDF( 0.7 , 0.30000000000000004 )= 4.206875021598653e-05
CDF( 0.8 , 0.19999999999999996 )= 2.2650283596193056e-06
CDF( 0.9 , 0.09999999999999998 )= 1.160402191701829e-08
CDF( 1.0 , 0.0 )= 0.0
```
and

![cdf](https://github.com/user-attachments/assets/f57ef1e4-99ff-4d18-9c3e-1d819a5ef7ab)

Reply:
I dont have these values at all, did you set smalldimension to 0 to force monte carlo in this 2-d case ?
`ot.ResourceMap.SetAsUnsignedInteger("PointConditionalDistribution-SmallDimension", 0)`

Reply:
I played a little bit with the different multivariate integration algorithms we have in OT to compute the CDF. Here is the script:
```python
```

For a nice case (KernelMixture in dimension 3, Normal kernel), I get:
```
range=
 ]-inf (-9.43859), (9.7703) +inf[
]-inf (-9.83201), (9.89686) +inf[
]-inf (-9.54297), (9.50642) +inf[
CDF along the diagonal
========================================================================
[IteratedQuadrature] CDF( 0.0 ,... , 0.0 )=0.15155223095112888 t=0.112 s
[GaussLegendre     ] CDF( 0.0 ,... , 0.0 )=0.15155223095086195 t=2.23 s
[SimplicialCubature] CDF( 0.0 ,... , 0.0 )=0.15155223094637499 t=3.86 s
[Cuba cuhre        ] CDF( 0.0 ,... , 0.0 )=0.15155223095073478 t=0.303 s
[Cuba suave        ] CDF( 0.0 ,... , 0.0 )=0.15151372708838273 t=0.338 s
[Cuba vegas        ] CDF( 0.0 ,... , 0.0 )=0.15157319907812616 t=0.285 s
[Cuba divonne      ] CDF( 0.0 ,... , 0.0 )=0.15157284077517524 t=0.393 s
========================================================================
[IteratedQuadrature] CDF( 0.1 ,... , 0.1 )=0.17937736905241017 t=0.213 s
[GaussLegendre     ] CDF( 0.1 ,... , 0.1 )=0.1793773690519932 t=2.22 s
[SimplicialCubature] CDF( 0.1 ,... , 0.1 )=0.17937736904698356 t=4.21 s
[Cuba cuhre        ] CDF( 0.1 ,... , 0.1 )=0.17937736905183796 t=0.304 s
[Cuba suave        ] CDF( 0.1 ,... , 0.1 )=0.1793157498777555 t=0.32 s
[Cuba vegas        ] CDF( 0.1 ,... , 0.1 )=0.17939785769705199 t=0.275 s
[Cuba divonne      ] CDF( 0.1 ,... , 0.1 )=0.17940127161037639 t=0.367 s
========================================================================
[IteratedQuadrature] CDF( 0.2 ,... , 0.2 )=0.21004497017699436 t=0.199 s
[GaussLegendre     ] CDF( 0.2 ,... , 0.2 )=0.21004497017652535 t=2.22 s
[SimplicialCubature] CDF( 0.2 ,... , 0.2 )=0.2100449701710787 t=4.3 s
[Cuba cuhre        ] CDF( 0.2 ,... , 0.2 )=0.21004497017306817 t=0.312 s
[Cuba suave        ] CDF( 0.2 ,... , 0.2 )=0.20991497076920734 t=0.326 s
[Cuba vegas        ] CDF( 0.2 ,... , 0.2 )=0.21007209200577531 t=0.277 s
[Cuba divonne      ] CDF( 0.2 ,... , 0.2 )=0.21007141250078878 t=0.388 s
========================================================================
[IteratedQuadrature] CDF( 0.3 ,... , 0.3 )=0.24341628375589702 t=0.263 s
[GaussLegendre     ] CDF( 0.3 ,... , 0.3 )=0.24341628375536301 t=2.23 s
[SimplicialCubature] CDF( 0.3 ,... , 0.3 )=0.24341628375028609 t=4.58 s
[Cuba cuhre        ] CDF( 0.3 ,... , 0.3 )=0.24341628375070742 t=0.299 s
[Cuba suave        ] CDF( 0.3 ,... , 0.3 )=0.24334387043723193 t=0.336 s
[Cuba vegas        ] CDF( 0.3 ,... , 0.3 )=0.24344598561421132 t=0.287 s
[Cuba divonne      ] CDF( 0.3 ,... , 0.3 )=0.2434447387757292 t=0.398 s
========================================================================
[IteratedQuadrature] CDF( 0.4 ,... , 0.4 )=0.27927003290903296 t=0.228 s
[GaussLegendre     ] CDF( 0.4 ,... , 0.4 )=0.27927003290846214 t=2.24 s
[SimplicialCubature] CDF( 0.4 ,... , 0.4 )=0.27927003290354718 t=4.91 s
[Cuba cuhre        ] CDF( 0.4 ,... , 0.4 )=0.27927003291268426 t=0.316 s
[Cuba suave        ] CDF( 0.4 ,... , 0.4 )=0.27910793428168779 t=0.331 s
[Cuba vegas        ] CDF( 0.4 ,... , 0.4 )=0.2792962103307905 t=0.288 s
[Cuba divonne      ] CDF( 0.4 ,... , 0.4 )=0.27930023487617661 t=0.401 s
========================================================================
[IteratedQuadrature] CDF( 0.5 ,... , 0.5 )=0.31730477853373501 t=0.263 s
[GaussLegendre     ] CDF( 0.5 ,... , 0.5 )=0.31730477853311589 t=2.25 s
[SimplicialCubature] CDF( 0.5 ,... , 0.5 )=0.3173047785263774 t=5.19 s
[Cuba cuhre        ] CDF( 0.5 ,... , 0.5 )=0.31730477852302946 t=0.304 s
[Cuba suave        ] CDF( 0.5 ,... , 0.5 )=0.31711857533250787 t=0.321 s
[Cuba vegas        ] CDF( 0.5 ,... , 0.5 )=0.31733737875614515 t=0.274 s
[Cuba divonne      ] CDF( 0.5 ,... , 0.5 )=0.31733548094958119 t=0.375 s
========================================================================
[IteratedQuadrature] CDF( 0.6 ,... , 0.6 )=0.35714626732001103 t=0.235 s
[GaussLegendre     ] CDF( 0.6 ,... , 0.6 )=0.35714626731932747 t=2.25 s
[SimplicialCubature] CDF( 0.6 ,... , 0.6 )=0.35714626731314358 t=5.61 s
[Cuba cuhre        ] CDF( 0.6 ,... , 0.6 )=0.35714626732137694 t=0.3 s
[Cuba suave        ] CDF( 0.6 ,... , 0.6 )=0.35687356104093182 t=0.331 s
[Cuba vegas        ] CDF( 0.6 ,... , 0.6 )=0.35717296378183255 t=0.293 s
[Cuba divonne      ] CDF( 0.6 ,... , 0.6 )=0.35717204383549483 t=0.378 s
========================================================================
[IteratedQuadrature] CDF( 0.7 ,... , 0.7 )=0.39835957118825899 t=0.211 s
[GaussLegendre     ] CDF( 0.7 ,... , 0.7 )=0.39835957118736864 t=2.25 s
[SimplicialCubature] CDF( 0.7 ,... , 0.7 )=0.39835957118051657 t=5.93 s
[Cuba cuhre        ] CDF( 0.7 ,... , 0.7 )=0.39835957119071813 t=0.302 s
[Cuba suave        ] CDF( 0.7 ,... , 0.7 )=0.39838253549785396 t=0.345 s
[Cuba vegas        ] CDF( 0.7 ,... , 0.7 )=0.39840000275600457 t=0.292 s
[Cuba divonne      ] CDF( 0.7 ,... , 0.7 )=0.39838619112138918 t=0.403 s
========================================================================
[IteratedQuadrature] CDF( 0.8 ,... , 0.8 )=0.44046528649587013 t=0.203 s
[GaussLegendre     ] CDF( 0.8 ,... , 0.8 )=0.44046528649484251 t=2.24 s
[SimplicialCubature] CDF( 0.8 ,... , 0.8 )=0.44046528648757222 t=6.29 s
[Cuba cuhre        ] CDF( 0.8 ,... , 0.8 )=0.44046528649534805 t=0.293 s
[Cuba suave        ] CDF( 0.8 ,... , 0.8 )=0.44027807206254799 t=0.321 s
[Cuba vegas        ] CDF( 0.8 ,... , 0.8 )=0.44051171846556098 t=0.28 s
[Cuba divonne      ] CDF( 0.8 ,... , 0.8 )=0.44049189285677259 t=0.428 s
========================================================================
[IteratedQuadrature] CDF( 0.9 ,... , 0.9 )=0.48295859946902558 t=0.278 s
[GaussLegendre     ] CDF( 0.9 ,... , 0.9 )=0.48295859946784347 t=2.24 s
[SimplicialCubature] CDF( 0.9 ,... , 0.9 )=0.48295859946079683 t=6.58 s
[Cuba cuhre        ] CDF( 0.9 ,... , 0.9 )=0.48295859950786951 t=0.298 s
[Cuba suave        ] CDF( 0.9 ,... , 0.9 )=0.48281744667676141 t=0.325 s
[Cuba vegas        ] CDF( 0.9 ,... , 0.9 )=0.48299376510430642 t=0.275 s
[Cuba divonne      ] CDF( 0.9 ,... , 0.9 )=0.48297869776895591 t=0.371 s
========================================================================
[IteratedQuadrature] CDF( 1.0 ,... , 1.0 )=0.52532971735717549 t=0.23 s
[GaussLegendre     ] CDF( 1.0 ,... , 1.0 )=0.52532971735590084 t=2.22 s
[SimplicialCubature] CDF( 1.0 ,... , 1.0 )=0.52532971734690515 t=7.07 s
[Cuba cuhre        ] CDF( 1.0 ,... , 1.0 )=0.52532971739542889 t=0.292 s
[Cuba suave        ] CDF( 1.0 ,... , 1.0 )=0.52519017457114703 t=0.33 s
[Cuba vegas        ] CDF( 1.0 ,... , 1.0 )=0.52538079578033259 t=0.279 s
[Cuba divonne      ] CDF( 1.0 ,... , 1.0 )=0.52534808520099119 t=0.386 s
```
We see that the deterministic algorithms (the 4 first algorithms) agree on the value and are very accurate (~10 digits), while the random algorithms (the 3 last algorithms) are moderately accurate (~3-4 digits)

For an intermediate difficult case (NormalCopula in dimension 3) I get:
```
range=
 [0, 1]
[0, 1]
[0, 1]
CDF along the diagonal
========================================================================
[IteratedQuadrature] CDF( 0.0 ,... , 0.0 )=0 t=0.0697 s
[GaussLegendre     ] CDF( 0.0 ,... , 0.0 )=0 t=2.86e-05 s
[SimplicialCubature] CDF( 0.0 ,... , 0.0 )=0 t=0.0832 s
[Cuba cuhre        ] CDF( 0.0 ,... , 0.0 )=0 t=0.19 s
[Cuba suave        ] CDF( 0.0 ,... , 0.0 )=0 t=0.206 s
[Cuba vegas        ] CDF( 0.0 ,... , 0.0 )=0 t=0.175 s
[Cuba divonne      ] CDF( 0.0 ,... , 0.0 )=0 t=0.196 s
========================================================================
[IteratedQuadrature] CDF( 0.1 ,... , 0.1 )=0.00021178395551085748 t=7.76 s
[GaussLegendre     ] CDF( 0.1 ,... , 0.1 )=0.00021178248924816071 t=9.27 s
[SimplicialCubature] CDF( 0.1 ,... , 0.1 )=0.00021178241522188597 t=11.1 s
[Cuba cuhre        ] CDF( 0.1 ,... , 0.1 )=0.00021178228248419734 t=0.328 s
[Cuba suave        ] CDF( 0.1 ,... , 0.1 )=0.00021166813161643857 t=0.34 s
[Cuba vegas        ] CDF( 0.1 ,... , 0.1 )=0.0002117867723710615 t=0.334 s
[Cuba divonne      ] CDF( 0.1 ,... , 0.1 )=0.00021178458832134009 t=0.368 s
========================================================================
[IteratedQuadrature] CDF( 0.2 ,... , 0.2 )=0.0020511826701073953 t=52.3 s
[GaussLegendre     ] CDF( 0.2 ,... , 0.2 )=0.0020511831413096587 t=9.57 s
[SimplicialCubature] CDF( 0.2 ,... , 0.2 )=0.0020511826484399358 t=11.4 s
[Cuba cuhre        ] CDF( 0.2 ,... , 0.2 )=0.0020511811802552205 t=0.345 s
[Cuba suave        ] CDF( 0.2 ,... , 0.2 )=0.0020499142166379731 t=0.385 s
[Cuba vegas        ] CDF( 0.2 ,... , 0.2 )=0.0020512248570605722 t=0.351 s
[Cuba divonne      ] CDF( 0.2 ,... , 0.2 )=0.0020512119680529384 t=0.382 s
========================================================================
[IteratedQuadrature] CDF( 0.3 ,... , 0.3 )=0.0081140246871726773 t=105 s
[GaussLegendre     ] CDF( 0.3 ,... , 0.3 )=0.0081140315427968498 t=9.05 s
[SimplicialCubature] CDF( 0.3 ,... , 0.3 )=0.0081140301070055723 t=10.9 s
[Cuba cuhre        ] CDF( 0.3 ,... , 0.3 )=0.0081140243641324473 t=0.346 s
[Cuba suave        ] CDF( 0.3 ,... , 0.3 )=0.0081133185762635486 t=0.355 s
[Cuba vegas        ] CDF( 0.3 ,... , 0.3 )=0.0081142093679498137 t=0.329 s
[Cuba divonne      ] CDF( 0.3 ,... , 0.3 )=0.0081141784437054488 t=0.372 s
========================================================================
[IteratedQuadrature] CDF( 0.4 ,... , 0.4 )=0.022238371753678012 t=141 s
[GaussLegendre     ] CDF( 0.4 ,... , 0.4 )=0.02223839131152441 t=9.05 s
[SimplicialCubature] CDF( 0.4 ,... , 0.4 )=0.02223838841987049 t=10.8 s
[Cuba cuhre        ] CDF( 0.4 ,... , 0.4 )=0.022238373516567263 t=0.315 s
[Cuba suave        ] CDF( 0.4 ,... , 0.4 )=0.022231546825327742 t=0.356 s
[Cuba vegas        ] CDF( 0.4 ,... , 0.4 )=0.022238910264702299 t=0.317 s
[Cuba divonne      ] CDF( 0.4 ,... , 0.4 )=0.022238875720750662 t=0.368 s
========================================================================
[IteratedQuadrature] CDF( 0.5 ,... , 0.5 )=0.049960058626775314 t=182 s
[GaussLegendre     ] CDF( 0.5 ,... , 0.5 )=0.049960101182717938 t=8.95 s
[SimplicialCubature] CDF( 0.5 ,... , 0.5 )=0.049960096660057442 t=10.8 s
[Cuba cuhre        ] CDF( 0.5 ,... , 0.5 )=0.049960065159866736 t=0.338 s
[Cuba suave        ] CDF( 0.5 ,... , 0.5 )=0.049946653895524197 t=0.36 s
[Cuba vegas        ] CDF( 0.5 ,... , 0.5 )=0.049961358363829984 t=0.313 s
[Cuba divonne      ] CDF( 0.5 ,... , 0.5 )=0.049961490785760697 t=0.366 s
========================================================================
[IteratedQuadrature] CDF( 0.6 ,... , 0.6 )=0.099342543004688214 t=227 s
[GaussLegendre     ] CDF( 0.6 ,... , 0.6 )=0.099342621045210855 t=9.49 s
[SimplicialCubature] CDF( 0.6 ,... , 0.6 )=0.099342615293214295 t=11.2 s
[Cuba cuhre        ] CDF( 0.6 ,... , 0.6 )=0.099342557268207704 t=0.338 s
[Cuba suave        ] CDF( 0.6 ,... , 0.6 )=0.099331761261356635 t=0.349 s
[Cuba vegas        ] CDF( 0.6 ,... , 0.6 )=0.099345366733025381 t=0.308 s
[Cuba divonne      ] CDF( 0.6 ,... , 0.6 )=0.09934624195429767 t=0.367 s
========================================================================
[IteratedQuadrature] CDF( 0.7 ,... , 0.7 )=0.18257758717131667 t=247 s
[GaussLegendre     ] CDF( 0.7 ,... , 0.7 )=0.18257771687194252 t=9.26 s
[SimplicialCubature] CDF( 0.7 ,... , 0.7 )=0.18257771073695633 t=11 s
[Cuba cuhre        ] CDF( 0.7 ,... , 0.7 )=0.18257761365280339 t=0.343 s
[Cuba suave        ] CDF( 0.7 ,... , 0.7 )=0.18255739235799553 t=0.385 s
[Cuba vegas        ] CDF( 0.7 ,... , 0.7 )=0.18258327288343518 t=0.332 s
[Cuba divonne      ] CDF( 0.7 ,... , 0.7 )=0.18258728016273176 t=0.354 s
========================================================================
[IteratedQuadrature] CDF( 0.8 ,... , 0.8 )=0.31955132447068918 t=286 s
[GaussLegendre     ] CDF( 0.8 ,... , 0.8 )=0.3195515267376674 t=9.57 s
[SimplicialCubature] CDF( 0.8 ,... , 0.8 )=0.31955152214302773 t=11.2 s
[Cuba cuhre        ] CDF( 0.8 ,... , 0.8 )=0.3195513732041364 t=0.338 s
[Cuba suave        ] CDF( 0.8 ,... , 0.8 )=0.31947944911246717 t=0.37 s
[Cuba vegas        ] CDF( 0.8 ,... , 0.8 )=0.31955709883800176 t=0.347 s
[Cuba divonne      ] CDF( 0.8 ,... , 0.8 )=0.31957426514361753 t=0.367 s
========================================================================
[IteratedQuadrature] CDF( 0.9 ,... , 0.9 )=0.54826293941936266 t=305 s
[GaussLegendre     ] CDF( 0.9 ,... , 0.9 )=0.54826323767779561 t=9.51 s
[SimplicialCubature] CDF( 0.9 ,... , 0.9 )=0.54826324599724696 t=11.2 s
[Cuba cuhre        ] CDF( 0.9 ,... , 0.9 )=0.54826314818542632 t=0.347 s
[Cuba suave        ] CDF( 0.9 ,... , 0.9 )=0.54799170673678166 t=0.445 s
[Cuba vegas        ] CDF( 0.9 ,... , 0.9 )=0.54826353622782908 t=0.454 s
[Cuba divonne      ] CDF( 0.9 ,... , 0.9 )=0.54831225246457782 t=0.34 s
========================================================================
```
This time, the deterministic methods are better than the stochastic ones but only with a slight margin. And yes, for x==1 I am still waiting for IteratedQuadrature to finish after more than one hour. The best choice seems to be cuhre for the low dimensions, I will perform more tests on different dimensions soon.